embulk-input-s3/src/main/java/org/embulk/input/s3/AbstractS3FileInputPlugin.java

```diff
@@ -67,6 +67,11 @@
         @ConfigDefault("null")
         public Optional<String> getAccessKeyId();
 
+        @Config("http_proxy")
+        @ConfigDefault("null")
+        public Optional<HttpProxy> getHttpProxy();
+        public void setHttpProxy(Optional<HttpProxy> httpProxy);
```

Comment:
I don't think we should introduce new standards in environment variables. No one will get benefits with the new env.

----

It's still a design choice we should have a kind of "default" envs to use. I'm personally -1 on defaults. My idea is just to receive a new `env` configuration such as :

```
http_proxy:
  - env: [ "HTTP_PROXY", "http_proxy", ... ]  # First match
  # Exclusive from host, port, ...
```

Even if it has a kind of default like below, the benefit of users are very small. Users save just ~10 key types in making a configuration. Is it really a benefit?

```
  - env: []  # Use default
```
```
  - env_default: True
```

I believe users achieve more benefits by the proxy configuration is always explicit and clear. They won't be confused any implicit configurations they're unaware.